### DISCUSSION TITLE
Looking for help

### DISCUSSION THREAD

**Discussion by perry-mitchell** [buttercup/buttercup-desktop#1224] [Announcements]
Buttercup needs help: Don't worry, we're not in any trouble or at risk of "shutting down" - whatever that'd mean for an open source Github repo. But we are stagnating a bit.. due to the overwhelmingly large amount of work we have to get the platform into shape, add new awesome features, etc.. and we could really use a hand.

We already get a fair bit of help from the community in terms of reported issues, constant QA and the occasional PR.. but we'd ideally have a couple more regular developers, like myself, working on the platform to take it to the next level.

Before I continue; I'm not asking anyone to commit a ton of time for free - I want Buttercup to be a long-term, well known, trusted password manager for the years to come. I _do_ want a business out of it, but I want that so I can remain full time on this and all of the open source production we've been trying to support already. I'm looking for people that want to be a part of the team, potentially take some larger role in the company, and hopefully eventually be hired to continue publishing the free and open software you see here.

I'm remarkably bad at sourcing help (asking for it, and knowing where to find it).. but Buttercup's community is awesome and I know there are some of you out there that might want such a challenge.. to be a part of this going forward. All I'm asking for is a chat :)

**Comment by subdavis** [comment]
Hey there!  I think we've crossed paths before (perhaps related to [kdbxweb](https://github.com/keeweb/kdbxweb)?) and this post struck a bit of a chord with me because I remember feeling this way too nearly 4 years ago.

Open source is really hard.  Even moderately successful projects like buttercup struggle, and the open-source-password-manager space is exceedingly crowded.  For a brief few years, I was the primary maintainer of [Tusk](https://github.com/subdavis/Tusk) for keepass, and the experience taught me something valuable.  Tusk had more users than Buttercup, but it still ultimately failed to attract enough maintainer support to carry on or attract enough users to generate any money to pay anyone.

The thing I learned is that only die-hard hobbyists care about trying to use an overly complicated technology like WebDAV or Google Drive to sync their passwords.  Most people want the Bitwarden experience, a totally integrated ecosystem with backend services and sharing.

It's a simple fact that KeeWeb is dead. The whole keepass ecosystem is either dead or dying.  It was never going to be an economically sustainable project and the software developers who take care of keepass do it for fun and love because there's just not a big enough market of potential users to ever support it.

Even passwords themselves are dying (OAuth, SSO, passwordless login) and the idea of a password manager is beginning to change.

> I do want a business out of it.

I think you've seen that there's no future in trying to carve off market share from Keepass or Unix Password Store or those other niche projects.  There's no future in trying to market Buttercup to highly technical hobby users like you and I. 

I see on your roadmap that you're beginning to think about managed services.  That's great.  Bitwarden basically has no major competitors right now in the world of open source.  If you're looking to get more engineers over here, you should focus on one big thing: what will make Buttercup different from Bitwarden?

I think if you'd done that from the beginning, you'd be in a different place now.  .NET totally sucks. Buttercup could have been the most popular, contributor-friendly password manager.

I wish you and the Buttercup project the best of luck.

**Reply by perry-mitchell** [reply]
> Most people want the Bitwarden experience, a totally integrated ecosystem with backend services and sharing.

You're right, of course. I think this is ultimately where Buttercup is heading, especially if it's to attract enough attention to have a chance at building a viable income stream.

At this point I think we're probably still going to invest a lot of time into the technical side of things, with Buttercup, such as it's cli tool, and other dev- related components that could make Buttercup a useful tool in development spaces etc.. Buttercup definitely does need to bring something different to the table if we're going to come up against Bitwarden and the other players in this space and these are challenges that will need solving once we have enough help to do so :)

> It's a simple fact that KeeWeb is dead. The whole keepass ecosystem is either dead or dying. It was never going to be an economically sustainable project and the software developers who take care of keepass do it for fun and love because there's just not a big enough market of potential users to ever support it.

I could see it heading this way when I was still using Keepass. It's super unfortunate because of how flexible Keepass was in some ways.. and the effort that's gone into it. It did lack cohesion in my opinion, with clients all over the place.. But it has been a while since I used it so I'll refrain from further speculation. It's unfortunate though.. I'd be sad to see it die.

> I think you've seen that there's no future in trying to carve off market share from Keepass or Unix Password Store or those other niche projects. There's no future in trying to market Buttercup to highly technical hobby users like you and I.

Indeed. I don't want to forsake any of the work that we've done so far in the "hobbyist" space, but it is time to expand it in a direction that's somehow more mainstream. 

> I think if you'd done that from the beginning, you'd be in a different place now. .NET totally sucks. Buttercup could have been the most popular, contributor-friendly password manager.

Well, we're still aiming for that. I think that it being entirely JavaScript and web technologies is a benefit for us in that regard. The tech is definitely good enough to build a password manager around but we maintain a very versatile codebase that can in fact run anywhere without compilation for different platforms. It's solely this that has allowed me to mostly solo develop parts of the platform for so long.

> I wish you and the Buttercup project the best of luck

Thanks! I appreciate it.
